wave_lang/kernel/wave/wave.py

```diff
@@ -567,18 +569,64 @@ def compile_to_mlir(
             llvm_func_config,
         )
 
-        emitter = WaveEmitter(
-            dispatch_entrypoint, trace, self.constraints, options, self.grid_type
-        )
-        try:
-            emitter.emit(trace.get_root_graph())
-        except:
-            logger.info("Error in emitter")
-            asm = mb.module_op.get_asm()
-            logger.info(asm)
-            raise
-        emitter.finish()
+        print("AFTER ", mb.module_op.get_asm())
+        if not module_op:
+            emitter = WaveEmitter(
+                dispatch_entrypoint, trace, self.constraints, options, self.grid_type
+            )
+            try:
+                emitter.emit(trace.get_root_graph())
+            except:
+                logger.info("Error in emitter")
+                asm = mb.module_op.get_asm()
+                logger.info(asm)
+                raise
+            emitter.finish()
+        else:
+            with exe._loc, InsertionPoint.at_block_begin(dispatch_entrypoint.entry_block):
+                target_block = dispatch_entrypoint.entry_block
+                source_func_op = list(module_op.operation.regions[0].blocks[0])[0]
```

Comment:
Nit: isn't there something like `.operartions` on a block that avoids materializing the entire list?